testdata/sqllogictests/functions/read_csv.slt

```diff
@@ -74,3 +80,26 @@ select * from read_csv('./testdata/csv/delimiter.csv', delimiter => ';');
 # Invalid delimiter (longer than one byte)
 statement error delimiters for CSV must fit in one byte \(e.g. ','\)
 select * from read_csv('./testdata/csv/delimiter.csv', delimiter => ';;');
+
+# Headers
+
+query ITTR rowsort
+select * from read_csv('./testdata/csv/headerless.csv', header => false);
+----
+1   hello   world   3.9
+2   HELLO   WORLD   4.9
+
+query ITTR rowsort
+select * from read_csv('./testdata/csv/headerless.csv', header => 'false');
```

Comment:
i don't think we should do implicit conversions. I'd expect any string value to error out here.

Reply:
I disagree with this, I think it's reasonable to to expect 'false' gets automatically cast to boolean here. And this auto cast feels consistent with other sql systems.

Postgres will convert 'false' to a boolean if the context expects a boolean: https://www.postgresql.org/docs/current/datatype-boolean.html

Reply:
fair enough.